services/src/main/java/org/keycloak/services/resources/admin/IdentityProvidersResource.java

```diff
@@ -166,23 +169,75 @@ public Map<String, String> importFrom(@Parameter(description = "JSON body") Map<
     }
 
     /**
-     * Get identity providers
+     * List identity providers.
      *
-     * @return
+     * @param search Filter to search specific providers by name. Search can be prefixed (name*), contains (*name*) or exact (\"name\"). Default prefixed.
+     * @param firstResult Pagination offset
+     * @param maxResults Maximum results size (defaults to 100)
+     * @return The list of providers.
      */
     @GET
     @Path("instances")
     @NoCache
     @Produces(MediaType.APPLICATION_JSON)
     @Tag(name = KeycloakOpenAPI.Admin.Tags.IDENTITY_PROVIDERS)
-    @Operation( summary = "Get identity providers")
-    public Stream<IdentityProviderRepresentation> getIdentityProviders() {
+    @Operation(summary = "List identity providers")
+    public Stream<IdentityProviderRepresentation> getIdentityProviders(
+            @Parameter(description = "Filter specific providers by name. Search can be prefix (name*), contains (*name*) or exact (\"name\"). Default prefixed.") @QueryParam("search") String search,
+            @Parameter(description = "Pagination offset") @QueryParam("first") Integer firstResult,
+            @Parameter(description = "Maximum results size (defaults to 100)") @QueryParam("max") Integer maxResults) {
         this.auth.realm().requireViewIdentityProviders();
 
+        if (firstResult == null) {
+            firstResult = 0;
+        }
+        if (maxResults == null) {
+            maxResults = 100;
+        }
+
         return realm.getIdentityProvidersStream()
+                .filter(predicateByName(search))
+                .skip(firstResult)
+                .limit(maxResults)
                 .map(provider -> StripSecretsUtils.strip(ModelToRepresentation.toRepresentation(realm, provider)));
     }
 
+    /**
+     * Count identity providers.
+     *
+     * @param search Filter to search specific providers by name. Search can be prefixed (name*), contains (*name*) or exact (\"name\"). Default prefixed.
+     * @return Integer count of providers.
+     */
+    @GET
+    @Path("instances/count")
+    @NoCache
+    @Produces(MediaType.APPLICATION_JSON)
+    @Tag(name = KeycloakOpenAPI.Admin.Tags.IDENTITY_PROVIDERS)
+    @Operation( summary = "Count identity providers")
+    public Integer countIdentityProviders(
```

Comment:
Why do we have a separate count endpoint? Under what circumstances should we be using this? If this a separate pre-flight that the client has to do to determine the pagination that is less than ideal as:

1. This complicates client logic for something we already fully know the state of on the server.
2. This state can become out of sync with reality as new records are inserted or deleted.
3. Because of nr. 2 we now need to do this request every time we also request the identity providers.
4. This results in poorer performance due to network overhead (although not significantly I imagine).
5. We cannot handle this generically in the client, as endpoints do not implement this consistently.